### DISCUSSION TITLE
reduce_topics with auto gives less count than the number of documents

### DISCUSSION THREAD

**Discussion by aminathshausan** [MaartenGr/BERTopic#1087] [Q&A]
Hi, 
I was fitting  a model with setting the  **nr_topics="auto"**  as below.  
My dataset has 1750 documents, but .get_topic_info() gives me total **Count** of 1627 documents including the outlier topic. 

`topic_model = BERTopic(verbose=True, nr_topics="auto")

topics, probs = topic_model.fit_transform(abstracts)

topic_desc = topic_model.get_topic_info()
`
**Questions**: 
1. Do the **Count** column from the **.get_topic_info()** represent the number of documents per topic? 
 2. If so, why is the total count not the same as the original document number? 
 3. If not, what do the Counts represent? 
 
Thank you

**Comment by MaartenGr** [comment]
The sum of the count column should be as large as the number of documents. If not, then something is likely going wrong. Could you share your code? Also, which version of BERTopic are you using?

**Reply by aminathshausan** [reply]
Thank you for the response. 
I used bertopic version 0.12.0, and the following code was used to fit the model. 

```
topic_model = BERTopic(verbose=True, nr_topics="auto", calculate_probabilities=True)

topics, probs = topic_model.fit_transform(abstracts)

topic_desc = topic_model.get_topic_info()
```

I am not sure what went wrong, such that the Count column did not add to total number of documents.
Thanks

**Reply by MaartenGr** [reply]
Could you also try it out with v0.14.0? I believe some things were fixed in between, so that might solve your issue.

**Reply by aminathshausan** [reply]
Thanks, Maarten.
I am using a Mac, and Jupyter Lab. 
My system currently has Python3.11. I tried to install bertopic version 0.14.0. But getting the following errors: 

```
ERROR: Cannot install bertopic because these package versions have conflicting dependencies.

The conflict is caused by:
    sentence-transformers 2.2.2 depends on torch>=1.6.0
    sentence-transformers 2.2.1 depends on torch>=1.6.0
    sentence-transformers 2.2.0 depends on torch>=1.6.0
    sentence-transformers 2.1.0 depends on torch>=1.6.0
    sentence-transformers 2.0.0 depends on torch>=1.6.0
    sentence-transformers 1.2.1 depends on torch>=1.6.0
    sentence-transformers 1.2.0 depends on torch>=1.6.0
    sentence-transformers 1.1.1 depends on torch>=1.6.0
    sentence-transformers 1.1.0 depends on torch>=1.6.0
    sentence-transformers 1.0.4 depends on torch>=1.6.0
    sentence-transformers 1.0.3 depends on torch>=1.6.0
    sentence-transformers 1.0.2 depends on torch>=1.6.0
    sentence-transformers 1.0.1 depends on torch>=1.6.0
    sentence-transformers 1.0.0 depends on torch>=1.6.0
    sentence-transformers 0.4.1.2 depends on torch>=1.6.0
    sentence-transformers 0.4.1.1 depends on torch>=1.6.0
    sentence-transformers 0.4.1 depends on torch>=1.6.0

To fix this you could try to:
1. loosen the range of package versions you've specified
2. remove package versions to allow pip attempt to solve the dependency conflict

ERROR: ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/topics/dependency-resolution/#dealing-with-dependency-conflicts
```
Following your recommendation in [this post](https://github.com/MaartenGr/BERTopic/discussions/967), I created a virtual environment with Python version 3.8, and tried to install bertopic 0.14.0. 

I am still getting the above errors. I am not sure what is going on. Could you please help. 
Thanks 

**Update**: I was able to install bertopic version 0.14.0, and ran the model. It now gives the correct number of documents in the Count column. 

Thank you so much for the help
